misc/keyvaluestorage/src/browser/index.ts

```diff
@@ -1,35 +1,66 @@
-import { safeJsonParse, safeJsonStringify } from "@walletconnect/safe-json";
-// @ts-expect-error
-import localStorage from "./lib/localStorage";
-
-import { IKeyValueStorage, parseEntry } from "../shared";
+import { IKeyValueStorage } from "../shared";
+import { IndexedDb } from "./lib/indexedDb";
+import { LocalStore } from "./lib/localStore";
+import { migrate } from "./lib/browserMigration";
 
 export class KeyValueStorage implements IKeyValueStorage {
-  private readonly localStorage: Storage = localStorage;
+  private storage: IKeyValueStorage;
+  private initialized = false;
+
+  constructor() {
+    const local = new LocalStore();
+    this.storage = local;
+    try {
+      const indexed = new IndexedDb();
+      migrate(local, indexed, this.isInitialized);
+      // indexedDb isn't available in node env so this will throw
+    } catch (e) {
+      this.initialized = true;
+    }
+  }
+
+  private isInitialized = (store: IKeyValueStorage) => {
```

Comment:
Nit on semantics: I was initially confused why a boolean was being passed as a callback function until I realised `isInitialized` is a method not a property:

```suggestion
  private setInitialized = (store: IKeyValueStorage) => {
```